pkg/client/unversioned/testclient/fixture.go

```diff
@@ -58,7 +58,14 @@ type ObjectScheme interface {
 func ObjectReaction(o ObjectRetriever, mapper meta.RESTMapper) ReactionFunc {
 
 	return func(action Action) (bool, runtime.Object, error) {
-		kind, err := mapper.KindFor(unversioned.GroupVersionResource{Resource: action.GetResource()})
+		var err error
+		var kind unversioned.GroupVersionKind
+
+		if action.GetSubresource() != "" {
+			kind, err = mapper.KindFor(unversioned.GroupVersionResource{Resource: action.GetSubresource()})
+		} else {
+			kind, err = mapper.KindFor(unversioned.GroupVersionResource{Resource: action.GetResource()})
+		}
 		if err != nil {
 			return false, nil, fmt.Errorf("unrecognized action %s: %v", action.GetResource(), err)
```

Comment:
With that change, it is sufficient to remove the TODO above regarding support for sub-resources.